cairo/ethereum/cancun/vm/instructions/log.cairo

```diff
@@ -0,0 +1,445 @@
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
+
+from ethereum.cancun.vm.stack import pop
+from ethereum.cancun.vm import Evm, EvmImpl
+from ethereum.cancun.vm.exceptions import ExceptionalHalt, WriteInStaticContext, OutOfGasError
+from ethereum.cancun.vm.memory import memory_read_bytes, expand_by
+from ethereum.cancun.vm.gas import calculate_gas_extend_memory, charge_gas, GasConstants
+from ethereum.utils.numeric import U256_to_be_bytes
+from ethereum_types.numeric import U256, U256Struct, Uint
+from ethereum_types.bytes import Bytes, Bytes32, Bytes32Struct, TupleBytes32, TupleBytes32Struct
+from ethereum_types.others import (
+    ListTupleU256U256,
+    ListTupleU256U256Struct,
+    TupleU256U256,
+    TupleU256U256Struct,
+)
+from ethereum.cancun.blocks import Log, LogStruct, TupleLog, TupleLogStruct
+
+// @notice LOG0 instruction - append log record with no topics
+func log0{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, evm: Evm}() -> ExceptionalHalt* {
+    alloc_locals;
+    const NUM_TOPICS = 0;
+
+    // STACK
+    let stack = evm.value.stack;
+    with stack {
+        // Pop memory_start_index and size
+        let (memory_start_index, err) = pop();
+        if (cast(err, felt) != 0) {
+            return err;
+        }
+        let (size, err) = pop();
+        if (cast(err, felt) != 0) {
+            return err;
+        }
+    }
+
+    // GAS
+    // Calculate memory expansion cost
+    // If the size is greater than 2**128, the memory expansion will trigger an out of gas error.
+    if (size.value.high != 0) {
+        tempvar err = new ExceptionalHalt(OutOfGasError);
+        return err;
+    }
+
+    // Calculate memory expansion cost
+    tempvar mem_access_tuple = new TupleU256U256(new TupleU256U256Struct(memory_start_index, size));
+    tempvar mem_access_list = ListTupleU256U256(new ListTupleU256U256Struct(mem_access_tuple, 1));
+    let extend_memory = calculate_gas_extend_memory(evm.value.memory, mem_access_list);
+
+    // Calculate total gas cost
+    let data_cost = GasConstants.GAS_LOG_DATA * size.value.low;
+    let topic_cost = GasConstants.GAS_LOG_TOPIC * NUM_TOPICS;
+    let total_cost = Uint(
+        GasConstants.GAS_LOG + data_cost + topic_cost + extend_memory.value.cost.value
+    );
+
+    let err = charge_gas(total_cost);
+    if (cast(err, felt) != 0) {
+        return err;
+    }
+
+    // OPERATION
+    let memory = evm.value.memory;
+    with memory {
+        expand_by(extend_memory.value.expand_by);
+        let data = memory_read_bytes(memory_start_index, size);
+    }
+
+    // Create log entry
+    let null_ptr = cast(0, Bytes32Struct*);
+    tempvar topics = TupleBytes32(new TupleBytes32Struct(new Bytes32(null_ptr), 0));
```

Comment:
wasn't 
```suggestion
    tempvar topics = TupleBytes32(new TupleBytes32Struct(cast(0, Bytes32Struct*), 0));
```

enough?

Reply:
I thought it was starting to be a bit heavy: 
```
tempvar topics = TupleBytes32(new TupleBytes32Struct(new Bytes32(cast(0, Bytes32Struct*)), 0));
```